packages/core-utils/src/optimism/batch-encoding.ts

```diff
@@ -56,72 +126,122 @@ const encodeBatchContext = (context: BatchContext): string => {
 }
 
 export const decodeAppendSequencerBatch = (
-  b: string
+  b: string | Buffer,
+  opts?: DecodeSequencerBatchOpts
 ): AppendSequencerBatchParams => {
-  b = remove0x(b)
+  let buf: Buffer
+  if (typeof b === 'string') {
+    b = remove0x(b)
+    buf = Buffer.from(b, 'hex')
+  } else {
+    buf = b
+  }
 
-  const shouldStartAtElement = b.slice(0, 10)
-  const totalElementsToAppend = b.slice(10, 16)
-  const contextHeader = b.slice(16, 22)
-  const contextCount = parseInt(contextHeader, 16)
+  const shouldStartAtElement = buf.slice(0, 5)
+  const totalElementsToAppend = buf.slice(5, 8)
+  const contextHeader = buf.slice(8, 11)
+  const contextCount = parseInt(contextHeader.toString('hex'), 8)
+  let batchType = BatchType.LEGACY
 
-  let offset = 22
-  const contexts = []
+  let offset = 11
+  let contexts = []
   for (let i = 0; i < contextCount; i++) {
-    const numSequencedTransactions = b.slice(offset, offset + 6)
-    offset += 6
-    const numSubsequentQueueTransactions = b.slice(offset, offset + 6)
-    offset += 6
-    const timestamp = b.slice(offset, offset + 10)
-    offset += 10
-    const blockNumber = b.slice(offset, offset + 10)
-    offset += 10
+    const numSequencedTransactions = buf.slice(offset, offset + 3)
+    offset += 3
+    const numSubsequentQueueTransactions = buf.slice(offset, offset + 3)
+    offset += 3
+    const timestamp = buf.slice(offset, offset + 5)
+    offset += 5
+    const blockNumber = buf.slice(offset, offset + 5)
+    offset += 5
     contexts.push({
-      numSequencedTransactions: parseInt(numSequencedTransactions, 16),
+      numSequencedTransactions: parseInt(
+        numSequencedTransactions.toString('hex'),
+        16
+      ),
       numSubsequentQueueTransactions: parseInt(
-        numSubsequentQueueTransactions,
+        numSubsequentQueueTransactions.toString('hex'),
         16
       ),
-      timestamp: parseInt(timestamp, 16),
-      blockNumber: parseInt(blockNumber, 16),
+      timestamp: parseInt(timestamp.toString('hex'), 16),
+      blockNumber: parseInt(blockNumber.toString('hex'), 16),
     })
   }
 
-  const transactions = []
+  if (contexts.length > 0) {
+    const context = contexts[0]
```

Comment:
This implementation is backwards

Reply:
The timestamp being 0 marks the existence of a dummy context and the blocknumber is the enum